lib/pattern.js

```diff
@@ -17,10 +17,11 @@ function Pattern(polys, opts) {
       svg.setAttribute('xmlns','http://www.w3.org/2000/svg');
     }
 
-    polys.forEach(function(poly) {
+    polys.forEach(function(poly, i) {
       var path = doc.createElementNS("http://www.w3.org/2000/svg", 'path');
       path.setAttribute("d", "M" + poly[1].join("L") + "Z");
       path.setAttribute("fill", poly[0]);
+      path.setAttribute("data-index", i);
```

Comment:
let's make this opt-in - can you put this behind `svgOpts.includePolyIndex` and add that option the the README under the SVG options?

i.e.
```
svgOpts.includePolyIndex && path.setAttribute("data-index", i);
```